exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/NestedLoopJoinTemplate.java

```diff
@@ -117,9 +117,12 @@ private int populateOutgoingBatch(JoinRelType joinType, int outputIndex) {
       for (; nextRightBatchToProcess < rightCounts.size(); nextRightBatchToProcess++) {
         int rightRecordCount = rightCounts.get(nextRightBatchToProcess);
         // for every record in right batch
+        final int currentRightBatchIndex = nextRightBatchToProcess << 16;
         for (; nextRightRecordToProcess < rightRecordCount; nextRightRecordToProcess++) {
 
-          if (doEval(nextLeftRecordToProcess, nextRightBatchToProcess, nextRightRecordToProcess)) {
+          // Since right container is a hyper container, in doEval generated code it expects the
```

Comment:
Minor: this comment ideally could be moved up to the statement where you do the left shift since that's the main change.